src/util.py

```diff
@@ -8,6 +9,49 @@
 settings = get_settings()
 
 
+class EmbedField(object):
+    name: str
+    value: str
+
+    def __init__(self, name, value):
+        self.name = name
+        self.value = value
+
+
+@asynccontextmanager
+async def create_embed(log_channel: discord.abc.GuildChannel, **kwargs):
+    # optional args
+    user = kwargs.get("user", None)
+    description = kwargs.get("description", None)
+    timestamp = kwargs.get("timestamp", None)
+    footer = kwargs.get("footer", None)
+    fields = kwargs.get("fields", None)
+
+    embed = discord.Embed()
+    try:
+        if user is not None:
+            embed.set_author(
+                name=user.display_name,
+                icon_url=user.display_avatar.url,
+            )
+        if description is not None:
+            embed.description = description
+        if timestamp is not None:
+            embed.timestamp = timestamp
+        if footer is not None:
+            embed.set_footer(text=footer)
+        if fields is not None:
+            for field in fields:
+                embed.add_field(name=field.name, value=field.value, inline=False)
+
+        yield embed
+    except Exception as e:
+        embed.add_field(name="Error", value=e, inline=False)
+        raise e
+    finally:
+        await log_channel.send(embed=embed)
```

Comment:
so the function name implies that we are only creating the embed and not sending it to the logging channel, which is two operations. so we can choose between two options
1. rename the function to be `create_and_send_embed`
2. return the embed here and leave the sending to the caller

Reply:
My 2 cents on a better name is `logging_embed_context` , since it's a context manager for the logging embed. Since every time we use this, we're going to want to be sending the embed to the logging channel, I'd rather not go with option 2